Intermittent test project build failures

## Summary

Since [this PR](https://github.com/dotnet/sdk/pull/41451), the test package projects are building on every build. There seems to be an intermittent issue in CI where the unit test projects themselves will fail to build because of the TestPackages scaffolding. In the error below, you can see that the `Microsoft.NET.Build.Containers.UnitTests.csproj` failed because of it trying to find `dotnet-portable` assets, which is strange. The relationship between the test package creation and the unit test project builds would need to be investigated.

```
D:\a\_work\1\s\.dotnet\sdk\9.0.100-preview.5.24307.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Publish.targets(355,5):
error MSB3030: Could not copy the file "D:\a\_work\1\s\src\Assets\TestPackages\dotnet-portable\obj\84f5ddf7-12f3-44b6-9898-5da1b99237c3.tmp" because it was not found.
[D:\a\_work\1\s\test\Microsoft.NET.Build.Containers.UnitTests\Microsoft.NET.Build.Containers.UnitTests.csproj::TargetFramework=net9.0]
```

## Binlog
https://microsoft-my.sharepoint.com/:u:/p/miyanni/EaZsT9xh-hVCu4h_1J2VonQBSJooCtr2KeSZcfahWBvIvg?e=vrq3Pa

--- a/test/Microsoft.NET.TestFramework/Microsoft.NET.TestFramework.csproj
+++ b/test/Microsoft.NET.TestFramework/Microsoft.NET.TestFramework.csproj
@@ -65,8 +65,8 @@
     <None Include="BuildTestPackages.targets" />
   </ItemGroup>
 
-  <Import Project="SetupTestRoot.targets" />
-  <Import Project="BuildTestPackages.targets" />
+  <Import Project="SetupTestRoot.targets" Condition="'$(BuildTestPackages)' != 'false'" />
+  <Import Project="BuildTestPackages.targets" Condition="'$(BuildTestPackages)' != 'false'" />
 
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
 

--- a/test/UnitTests.proj
+++ b/test/UnitTests.proj
@@ -116,7 +116,6 @@
     <Copy SourceFiles="@(AssetFiles)" DestinationFiles="@(AssetFiles->'$(HelixPayloadOnHost)\TestExecutionDirectoryFiles\Assets\%(RecursiveDir)%(Filename)%(Extension)')" />
   </Target>
 
-
   <Target Name="AppendHelixPreCommand" BeforeTargets="CoreTest" DependsOnTargets="CopyHelixFiles">
     <PropertyGroup>
       <HelixPreCommands Condition="!$(IsPosixShell)">call %HELIX_CORRELATION_PAYLOAD%\t\RunTestsOnHelix.cmd $(TestFullMSBuild);$(HelixPreCommands)</HelixPreCommands>
@@ -179,4 +178,5 @@
         HelixCorrelationPayload="@(HelixCorrelationPayload)"
         TestOutputDirectory="$(RepoRoot)artifacts\bin\localHelixTestLayout" />
   </Target>
+
 </Project>

--- a/test/xunit-runner/XUnitRunner.targets
+++ b/test/xunit-runner/XUnitRunner.targets
@@ -54,7 +54,7 @@
       <_CurrentRuntimeTargetFramework Condition="'$(_CurrentRuntimeTargetFramework)' == ''">$(SDKCustomXUnitRuntimeTargetFramework)</_CurrentRuntimeTargetFramework>
       <_CurrentAdditionalProperties>%(SDKCustomXUnitProject.AdditionalProperties)</_CurrentAdditionalProperties>
     </PropertyGroup>
-    <MSBuild Projects="$(_CurrentSDKCustomXUnitProject)" Targets="PublishWithOutput" Properties="CustomAfterMicrosoftCommonTargets=$(_SDKCustomXUnitPublishTargetsPath);TargetFramework=$(_CurrentPublishTargetFramework);$(_CurrentAdditionalProperties)">
+    <MSBuild Projects="$(_CurrentSDKCustomXUnitProject)" Targets="PublishWithOutput" Properties="CustomAfterMicrosoftCommonTargets=$(_SDKCustomXUnitPublishTargetsPath);TargetFramework=$(_CurrentPublishTargetFramework);$(BuildTestPackages)=false;$(_CurrentAdditionalProperties)">
       <Output TaskParameter="TargetOutputs" PropertyName="_PublishOutputDir" />
     </MSBuild>
     <MSBuild Projects="$(_CurrentSDKCustomXUnitProject)" Targets="GetTargetPath" Properties="CustomAfterMicrosoftCommonTargets=$(_SDKCustomXUnitPublishTargetsPath);TargetFramework=$(_CurrentPublishTargetFramework);$(_CurrentAdditionalProperties)">

--- a/test/xunit-runner/XUnitRunner.targets
+++ b/test/xunit-runner/XUnitRunner.targets
@@ -54,7 +54,7 @@
       <_CurrentRuntimeTargetFramework Condition="'$(_CurrentRuntimeTargetFramework)' == ''">$(SDKCustomXUnitRuntimeTargetFramework)</_CurrentRuntimeTargetFramework>
       <_CurrentAdditionalProperties>%(SDKCustomXUnitProject.AdditionalProperties)</_CurrentAdditionalProperties>
     </PropertyGroup>
-    <MSBuild Projects="$(_CurrentSDKCustomXUnitProject)" Targets="PublishWithOutput" Properties="CustomAfterMicrosoftCommonTargets=$(_SDKCustomXUnitPublishTargetsPath);TargetFramework=$(_CurrentPublishTargetFramework);$(BuildTestPackages)=false;$(_CurrentAdditionalProperties)">
+    <MSBuild Projects="$(_CurrentSDKCustomXUnitProject)" Targets="PublishWithOutput" Properties="CustomAfterMicrosoftCommonTargets=$(_SDKCustomXUnitPublishTargetsPath);TargetFramework=$(_CurrentPublishTargetFramework);BuildTestPackages=false;$(_CurrentAdditionalProperties)">
       <Output TaskParameter="TargetOutputs" PropertyName="_PublishOutputDir" />
     </MSBuild>
     <MSBuild Projects="$(_CurrentSDKCustomXUnitProject)" Targets="GetTargetPath" Properties="CustomAfterMicrosoftCommonTargets=$(_SDKCustomXUnitPublishTargetsPath);TargetFramework=$(_CurrentPublishTargetFramework);$(_CurrentAdditionalProperties)">

